packages/protocol/contracts/L1/hooks/AssignmentHook.sol

```diff
@@ -143,33 +145,45 @@ contract AssignmentHook is EssentialContract, IHook {
     /// @notice Hashes the prover assignment.
     /// @param _assignment The prover assignment.
     /// @param _taikoL1Address The address of the TaikoL1 contract.
+    /// @param _blockProposer The block proposer address.
+    /// @param _assignedProver The assigned prover address.
     /// @param _blobHash The blob hash.
     /// @return The hash of the prover assignment.
     function hashAssignment(
         ProverAssignment memory _assignment,
         address _taikoL1Address,
+        address _blockProposer,
+        address _assignedProver,
         bytes32 _blobHash
     )
         public
         view
         returns (bytes32)
     {
-        return keccak256(
+        bytes32 hash = keccak256(
             abi.encode(
-                "PROVER_ASSIGNMENT",
-                ITaikoL1(_taikoL1Address).getConfig().chainId,
-                _taikoL1Address,
-                address(this),
                 _assignment.metaHash,
                 _assignment.parentMetaHash,
-                _blobHash,
                 _assignment.feeToken,
                 _assignment.expiry,
                 _assignment.maxBlockId,
                 _assignment.maxProposedIn,
                 _assignment.tierFees
             )
         );
+
+        return keccak256(
+            abi.encodePacked(
+                "PROVER_ASSIGNMENT",
+                ITaikoL1(_taikoL1Address).getConfig().chainId,
+                _taikoL1Address,
+                _blockProposer,
```

Comment:
I see the recommendation from the bottom of the screenshot (currently cannot access the audit) to include the `_blockProposer` in the hash, but it is already part of it already indirectly via metaHash (since it is in BlockMetadata.sender) no ?

Reply:
Not really, since `assignment.metaHash` can be 0 if the assigned prover doesn't want to check metahash.